great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -130,43 +134,35 @@ def _query_for_year_and_month(
 
 
 def _get_sql_datetime_range(
-    conn: sqlalchemy.engine.base.Connection, table_name: str, col_name: str
+    conn: sqlalchemy.engine.base.Connection,
+    selectable: sqlalchemy.sql.Selectable,
+    col_name: str,
 ) -> DatetimeRange:
-    q = f"select min({col_name}), max({col_name}) from {table_name}"
-    min_max_dt = list(conn.execute(q))[0]
+    import sqlalchemy as sa
+
+    column = sa.column(col_name)
+    query = sa.select([func.min(col_name), func.max(column)]).select_from(selectable)
+    min_max_dt = list(conn.execute(query))[0]
+    if min_max_dt[0] is None or min_max_dt[1] is None:
+        raise SQLDatasourceError(
+            f"Data date range can not be determined for the query: {query}. The returned range was {min_max_dt}."
```

Comment:
@billdirks I believe that it is safer to explicitly convert `query` into `str`, for example:
```
query_string = query.compile(compile_kwargs={"literal_binds": True})
```

Thank you.

Reply:
As per earlier versions of `SQLAlchemy`, it has been utilized as a more explicit way to control what happens in "to string" operation.  It is certainly worth seeing how this approach has changed as the newer `SQLAlchemy` versions have been including more and more improvements.  Thanks.